shards/shards.go

```
@@ -580,98 +579,125 @@ func (ss *shardedSearcher) streamSearch(ctx context.Context, proc *process, q qu
-					metricSearchFilesSkippedTotal.Add(float64(sr.Stats.FilesSkipped))
-					metricSearchShardsSkippedTotal.Add(float64(sr.Stats.ShardsSkipped))
-					metricSearchMatchCountTotal.Add(float64(sr.Stats.MatchCount))
-					metricSearchNgramMatchesTotal.Add(float64(sr.Stats.NgramMatches))
-
-					// MaxPendingPriority *cannot* be this result's Priority, because
-					// the priority is removed before computing max() and calling sender.Send.
-					// (There may be duplicate priorities, though-- that's fine.) A PendingShard
-					// is one that has not entered this critical section and sent its results.
-					//
-					// Note that there are at least two layers above this implementing streamSearch
-					// or StreamSearch that also take a lock for the entirety of the Send() operation.
-					//
-					// This is to avoid a potential race between shards sending back results
-					// if the priority were removed before sending without a lock:
-					// 1) shard A (pri 1), B (pri 2), C (pri 3) dispatch, pendingPriorities = [1, 2, 3]
-					// 2) C completes and removes itself from the priority list, pP = [1, 2]
-					// 3) B completes, removes itself, computes max, *and sends results* as maxPendingPriority=1,
-					//    indicating that no future results will come from a lower-ordered shard, pP = [1]
-					// 4) A completes, removes itself, computes max, and sends results with maxPP=-Inf, indicating
-					//    that the stream is finished (?)
-					// 5) C finally wakes up, computes max, and sends results with maxPP=-Inf, but with priority=3.
-					mu.Lock()
-					pendingPriorities.remove(s.priority)
-					sr.Progress.MaxPendingPriority = pendingPriorities.max()
-					sr.Progress.Priority = s.priority
-					sender.Send(sr)
-					mu.Unlock()
-				}))
-				if err != nil {
-					mu.Lock()
-					pendingPriorities.remove(s.priority)
-					mu.Unlock()
-					return err
-				}
+	}
+
+search:
+	for {
+		_ = proc.Yield(ctx)
```

missing old comment on why we ignore error